pennylane/estimator/templates/comparators.py

```diff
@@ -0,0 +1,655 @@
+              y: ─├○────╭●─╰X─│───X─ x=y
+            |0>: ─╰X─╭●─│─────│───── x<y
+            |0>: ────╰X─╰X────╰X──── x>y
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects, where each object
+            represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        gate_lst = []
+
+        gate_lst.append(GateCount(resource_rep(qre.TemporaryAND), 1))
+        gate_lst.append(GateCount(resource_rep(qre.CNOT), 4))
+        gate_lst.append(GateCount(resource_rep(qre.X), 3))
+
+        return gate_lst
+
+
+class TwoQubitComparator(ResourceOperator):
+    r"""Resource class for comparing the integer values encoded in
+    two quantum registers of two qubits each.
+
+    This operation modifies the input registers. The original values can be restored
+    by applying the operation's adjoint.
+
+    Args:
+        wires (Sequence[int], None): the wires the operation acts on
+
+    Resources:
+        The resources are obtained from appendix B, Figure 3 in `arXiv:1711.10460
+        <https://arxiv.org/pdf/1711.10460>`_. Specifically,
+        the resources are given as :math:`2` ``CSWAP`` gates,
+        :math:`3` ``CNOT`` gates, and :math:`1` ``X`` gate. This decomposition
+        requires one zeroed auxiliary qubit.
+        The circuit which applies the comparison operation on registers :math:`(x0,x1)`
+        and :math:`(y0, y1)` is defined as:
+
+        .. code-block:: bash
+
+             x1 : ─╭X─╭●────╭●───────┤
+             y1 : ─╰●─│─────├SWAP────┤
+             x0 : ─╭X─├SWAP─│─────╭X─┤
+             y0 : ─╰●─│─────╰SWAP─╰●─┤
+            |1> : ────╰SWAP──────────┤
+
+    **Example**
+
+    The resources for this operation are computed using:
+
+    >>> two_qubit_compare = qre.TwoQubitComparator()
+    >>> print(qre.estimate(two_qubit_compare))
+    --- Resources: ---
+     Total qubits: 5
+     Total gates : 10
+     Qubit breakdown:
+      zeroed qubits: 1, any_state qubits: 0, algorithmic qubits: 4
+     Gate breakdown:
+      {'Toffoli': 2, 'CNOT': 7, 'X': 1}
+    """
+
+    num_wires = 4
+
+    def __init__(self, wires: WiresLike = None):
+        super().__init__(wires=wires)
+
+    @property
+    def resource_params(self):
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: An empty dictionary
+        """
+        return {}
+
+    @classmethod
+    def resource_rep(cls):
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute the resources.
+
+        Returns:
+            :class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`: the operator in a compressed representation
+        """
+        return CompressedResourceOp(cls, cls.num_wires, {})
+
+    @classmethod
+    def resource_decomp(cls, **kwargs):
```

Comment:
or here? If not, can we update the remaining cases of unnecessary kwargs?
```suggestion
    def resource_decomp(cls):
```